library(TwoSampleMR)
library(data.table)
library(MRPRESSO)

###AMA_exposure
setwd("E:/Rstudy/MRGWASdata/SEXdata")
a<-fread("GCST90029036_buildGRCh37.tsv",header = T)
b<-subset(a,p_value<5e-08)
names(b)[1:10]<-c("SNP","CHR","BP","effect allele","other allele","REF","eaf","beta","se","p")
write.csv(b,file="exposure_ama_raw.csv")
#Move the file "exposure_ama_raw.csv" to the folder where there is "TwoSampleMR" R package.
ama_raw<-system.file("MRstudydata/SEXcases/exposure_ama_raw.csv",package = "TwoSampleMR")
ama_exp_dat_clumped<-read_exposure_data(filename = ama_raw,sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",eaf_col = "eaf",clump = TRUE)
setwd("E:/Rstudy/MRGWASdata/ADdata")
c<-fread("AD_sumstats_Jansenetal_2019sept.txt",header = T)
names(c)[4:14]<-c("effect allele","other allele","SNP","Z","p","Nsum","Neff","dir","eaf","beta","se")
d<-merge(ama_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_ama_raw.csv")
outcome_dat<-read_outcome_data(snps =ama_exp_dat_clumped$SNP,filename = "outcome_ama_raw.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =ama_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_ama_raw.csv")
set.seed(1234)
mr_ama_raw<-generate_odds_ratios(mr_res = mr(dat))
presso_ama_raw<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                   OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                   SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###AMA (excluding AP4M1 variant rs999885, CELF1 variant rs11039266, and STXBP4 variant rs17817628)
excluded<-ama_exp_dat_clumped$SNP %in% c("rs999885","rs11039266","rs17817628")
ama_exclu_exp_dat_clumped<-ama_exp_dat_clumped[!excluded,]
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(ama_exclu_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_ama_exclu.csv")
outcome_dat<-read_outcome_data(snps =ama_exclu_exp_dat_clumped$SNP,filename = "outcome_ama_exclud.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =ama_exclu_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_ama_exclud.csv")
set.seed(1234)
mr_ama_exclu<-generate_odds_ratios(mr_res = mr(dat))
presso_ama_raw<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###AMO
setwd("E:/Rstudy/MRGWASdata/SEXdata")
a<-fread("repro_MENOPAUSE_AGE.sumstats.gz",header = T)
b<-subset(a,P<5e-08)
names(b)[4:10]<-c("other allele","effect allele","REF","eaf","beta","se","p")
write.csv(b,file="exposure_amo.csv")
#Move the file "exposure_amo.csv" to the folder where there is "TwoSampleMR" R package.
amo<-system.file("MRstudydata/SEXcases/exposure_amo.csv",package = "TwoSampleMR")
amo_exp_dat_clumped<-read_exposure_data(filename = amo,sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",eaf_col = "eaf",clump = TRUE)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(amo_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_amo.csv")
outcome_dat<-read_outcome_data(snps =amo_exp_dat_clumped$SNP,filename = "outcome_amo.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =amo_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_amo.csv")
set.seed(1234)
mr_amo<-generate_odds_ratios(mr_res = mr(dat))
presso_amo<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###AMO exclude the outliers including rs11586493, rs1979033
excluded<-amo_exp_dat_clumped$SNP %in% c("rs11586493","rs1979033")
amo_exclu_exp_dat_clumped<-amo_exp_dat_clumped[!excluded,]
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(amo_exclu_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_amo_exclu.csv")
outcome_dat<-read_outcome_data(snps =amo_exclu_exp_dat_clumped$SNP,filename = "outcome_amo_exclu.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =amo_exclu_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_amo_exclu.csv")
set.seed(1234)
mr_amo_exclu<-generate_odds_ratios(mr_res = mr(dat))
presso_amo_exclud<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###AMO excluded outliers and confunders
excluded<-amo_exclu_exp_dat_clumped$SNP %in% c("rs10516461","rs2277339","rs2077102","rs28880687","rs204883")
amo_exclu_exp_dat_clumped_01<-amo_exclu_exp_dat_clumped[!excluded,]
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(amo_exclu_exp_dat_clumped_01,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_amo_exclu_01.csv")
outcome_dat<-read_outcome_data(snps =amo_exclu_exp_dat_clumped_01$SNP,filename = "outcome_amo_exclu_01.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =amo_exclu_exp_dat_clumped_01,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_amo_exclu_01.csv")
set.seed(1234)
mr_amo_exclu<-generate_odds_ratios(mr_res = mr(dat))
presso_amo_exclu_01<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))


###AFS
setwd("E:/Rstudy/MRGWASdata/PREdata")
a<-fread("GCST90000045_buildGRCh37.tsv.gz",header = T)
b<-subset(a,p_value<5e-08)
names(b)[1:6]<-c("SNP","effect allele","other allele","beta","se","p")
write.csv(b,file="exposure_afs.csv")
#Move the file "exposure_afs.csv" to the folder where there is "TwoSampleMR" R package.
afs<-system.file("MRstudydata/SEXcases/exposure_afs.csv",package = "TwoSampleMR")
afs_exp_dat_clumped<-read_exposure_data(filename = afs,sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",eaf_col = "eaf",clump = TRUE)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(afs_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_afs.csv")
outcome_dat<-read_outcome_data(snps =afs_exp_dat_clumped$SNP,filename = "outcome_afs.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =afs_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_afs.csv")
set.seed(1234)
mr_afs<-generate_odds_ratios(mr_res = mr(dat))
presso_ama_raw<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###AFB
setwd("E:/Rstudy/MRGWASdata/PREdata")
a<-fread("GCST90000048_buildGRCh37.tsv.gz",header = T)
b<-subset(a,p_value<5e-08)
names(b)[1:6]<-c("SNP","effect allele","other allele","beta","se","p")
write.csv(b,file="exposure_afb.csv")
#Move the file "exposure_afb.csv" to the folder where there is "TwoSampleMR" R package.
afb<-system.file("MRstudydata/SEXcases/exposure_afb.csv",package = "TwoSampleMR")
afb_exp_dat_clumped<-read_exposure_data(filename = afb,sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",eaf_col = "eaf",clump = TRUE)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(afb_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_afb.csv")
outcome_dat<-read_outcome_data(snps =afb_exp_dat_clumped$SNP,filename = "outcome_afb.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =afb_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_afb.csv")
set.seed(1234)
mr<-generate_odds_ratios(mr_res = mr(dat))
presso<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###HRT
hrt_exp_dat_clumped<-extract_instruments(outcomes="ukb-b-18541",clump=TRUE,r2=0.001,kb=10000,p1=5e-08)
setwd("E:/Rstudy/MRGWASdata/ADdata")
c<-fread("AD_sumstats_Jansenetal_2019sept.txt",header = T)
d<-merge(hrt_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_hrt.csv")
outcome_dat<-read_outcome_data(snps =hrt_exp_dat_clumped$SNP,filename = "outcome_hrt.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =hrt_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_hrt.csv")
set.seed(1234)
mr<-generate_odds_ratios(mr_res = mr(dat))
presso<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
library(xlsx)
write.xlsx(hrt_exp_dat_clumped,"hrt_dat_clumpedre.xls")
write.xlsx(ORR,"ORR_hrtre.xls")
write.xlsx(mr,"mr_hrtre.xls")
write.xlsx(pleio,"pleio_hrtre.xls")
write.xlsx(heter,"heter_hrtre.xls")
write.xlsx(presso$`Main MR results`,"presso_Main MR results_hrtre.xls")
write.xlsx(presso$`MR-PRESSO results`,"presso_MR-PRESSO results_hrtre.xls")
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###HRT excluded rs11586493
excluded<-hrt_exp_dat_clumped$SNP %in% "rs11586493"
hrt_exclu_exp_dat_clumped<-hrt_exp_dat_clumped[!excluded,]
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(hrt_exclu_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_hrt_exclu.csv")
outcome_dat<-read_outcome_data(snps =hrt_exclu_exp_dat_clumped$SNP,filename = "outcome_hrt_exclu.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =hrt_exclu_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_hrt_exclu.csv")
set.seed(1234)
mr_hrt_exclu<-generate_odds_ratios(mr_res = mr(dat))
presso_ama_raw<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###NEB
neb_exp_dat_clumped<-extract_instruments(outcomes="ieu-b-4828",clump=TRUE,r2=0.001,kb=10000,p1=5e-06)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(neb_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_neb.csv")
outcome_dat<-read_outcome_data(snps =neb_exp_dat_clumped$SNP,filename = "outcome_neb.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =neb_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_neb.csv")
set.seed(1234)
mr<-generate_odds_ratios(mr_res = mr(dat))
presso<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###EOC
eoc_exp_dat_clumped<-extract_instruments(outcomes="ukb-a-322",clump=TRUE,r2=0.001,kb=10000,p1=5e-06)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(eoc_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_eoc.csv")
outcome_dat<-read_outcome_data(snps =eoc_exp_dat_clumped$SNP,filename = "outcome_eoc.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =eoc_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_eoc.csv")
set.seed(1234)
mr<-generate_odds_ratios(mr_res = mr(dat))
presso<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###ASOC
asoc_exp_dat_clumped<-extract_instruments(outcomes="ukb-b-9433",clump=TRUE,r2=0.001,kb=10000,p1=5e-08)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(asoc_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_asoc.csv")
outcome_dat<-read_outcome_data(snps =asoc_exp_dat_clumped$SNP,filename = "outcome_asoc.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =asoc_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_asoc.csv")
set.seed(1234)
mr<-generate_odds_ratios(mr_res = mr(dat))
presso_ama_raw<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))

###ALOC
aloc_exp_dat_clumped<-extract_instruments(outcomes="ukb-b-4206",clump=TRUE,r2=0.001,kb=10000,p1=5e-06)
setwd("E:/Rstudy/MRGWASdata/ADdata")
d<-merge(aloc_exp_dat_clumped,c,by.x = "SNP",by.y = "SNP")
write.csv(d,file = "outcome_aloc.csv")
outcome_dat<-read_outcome_data(snps =aloc_exp_dat_clumped$SNP,filename = "outcome_aloc.csv",sep = ",",snp_col = "SNP",beta_col = "beta",se_col = "se",effect_allele_col = "effect allele",other_allele_col = "other allele",pval_col = "p")
dat<-harmonise_data(exposure_dat =aloc_exp_dat_clumped,outcome_dat = outcome_dat)
write.csv(dat,file = "harmonize_dat_aloc.csv")
set.seed(1234)
mr<-generate_odds_ratios(mr_res = mr(dat))
presso_ama_raw<-mr_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", 
                          OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 5000,  
                          SignifThreshold = 0.05)
res_single<-mr_singlesnp(dat)
ORR<-generate_odds_ratios(res_single)
heter<-mr_heterogeneity(dat)
pleio<-mr_pleiotropy_test(dat)
set.seed(1234)
mr_scatter_plot(mr(dat),dat)
mr_forest_plot(res_single)
mr_funnel_plot(singlesnp_results = res_single)
mr_leaveoneout_plot(leaveoneout_results = mr_leaveoneout(dat))